### PR TITLE
`qml.simplify` can reduce number of rotation parameters

### PR DESCRIPTION
**Pull request by albi3ro**
We encountered a naming conflict between `qml.transforms.simplify`, which reduced the number of rotation parameters, and `qml.ops.functions.simplify`, which reduced arithmetic nesting.

To solve this problem, we make `qml.simplify`(`qml.ops.functions.simplify`) do both things.  This is accomplished by defining the `simplify` method for `Rot`, `CRot`, `U2`, and `U3` using the code from `pennylane/transforms/commutation_dag.py`.

Now we have both:
```pycon
>>>  op = qml.adjoint(qml.RX(0.54, wires=0) + qml.PauliX(0) + qml.PauliZ(1))
>>> qml.simplify(op)
(Adjoint(RX)(0.54, wires=[0]),
Adjoint(PauliX)(wires=[0]),
Adjoint(PauliZ)(wires=[1]))
>>> qml.simplify(qml.Rot(np.pi / 2, 0.1, -np.pi / 2, wires=0))
RX(0.1, wires=[0])
```

### CODE REVIEW

File: pennylane/transforms/commutation_dag.py [hunk 1]
```diff
@@ -1043,6 +896,7 @@ def size(self):
         """
         return len(self._multi_graph)
 
+    # pylint: disable=no-member
```
Comment (AlbertMitjans) [inline pennylane/transforms/commutation_dag.py:899]: Why is this needed?

**Review by AlbertMitjans — APPROVED** [review]
Good job!! 💯 🎉

**Review by rmoyard — COMMENTED** [review]
Great work @albi3ro, could you add an example where you have the sum or the product of rotations that will be fully simplified? Otherwise happy to approve it 💯

File: pennylane/transforms/commutation_dag.py (hunk 1 above)
Comment (albi3ro) [reply pennylane/transforms/commutation_dag.py:899]: I was getting a pylint warning about `dot.write_png`.

**Comment by rmoyard** [comment]
Awesome @albi3ro, thank you!

File: pennylane/transforms/commutation_dag.py (hunk 1 above)
Comment (AlbertMitjans) [reply pennylane/transforms/commutation_dag.py:899]: Could you maybe move it to the line where `dot.write_png` is? Doing this we avoid disabling the pylint check for the whole function.
